public/app/plugins/panel/geomap/layers/data/heatMap.tsx

```diff
@@ -0,0 +1,243 @@
+        }
+      },
+    };
+  },
+  // Heatmap overlay options
+  registerOptionsUI: (builder) => {
+    builder
+      .addSelect({
+        path: 'queryFormat.locationType',
+        name: 'Query Format',
+        defaultValue: defaultOptions.queryFormat.locationType,
+        settings: {
+          options: [
+            {
+              value: 'coordinates',
+              label: 'Coordinates',
+            },
+            {
+              value: 'geohash',
+              label: 'Geohash',
+            },
+          ],
+        },
+      })
+      .addFieldNamePicker({
+        path: 'fieldMapping.latitudeField',
+        name: 'Latitude Field',
+        defaultValue: defaultOptions.fieldMapping.latitudeField,
+        settings: {
+          filter: (f) => f.type === FieldType.number,
+          noFieldsMessage: 'No numeric fields found',
+        },
+        showIf: (config) =>
+          config.queryFormat.locationType === 'coordinates',
+      })
+      .addFieldNamePicker({
+        path: 'fieldMapping.longitudeField',
+        name: 'Longitude Field',
+        defaultValue: defaultOptions.fieldMapping.longitudeField,
+        settings: {
+          filter: (f) => f.type === FieldType.number,
+          noFieldsMessage: 'No numeric fields found',
+        },
+        showIf: (config) =>
+          config.queryFormat.locationType === 'coordinates',
+      })
+      .addFieldNamePicker({
+        path: 'fieldMapping.geohashField',
+        name: 'Geohash Field',
+        defaultValue: defaultOptions.fieldMapping.geohashField,
+        settings: {
+          filter: (f) => f.type === FieldType.string,
+          noFieldsMessage: 'No strings fields found',
+          info: ({
+            name,
+            field,
+          }) => {
+            if(!name || !field) {
+              return <div>Select a field that contains <a href="https://en.wikipedia.org/wiki/Geohash">geohash</a> values in each row.</div>
+            }
+            const first = reduceField({field, reducers:[ReducerID.firstNotNull]})[ReducerID.firstNotNull] as string;
+            if(!first) {
+              return <div>No values found</div>
+            }
+            // const coords = decodeGeohash(first);
+            // if(coords) {
+            //   return <div>First value: {`${coords}`} // {new Date().toISOString()}</div>
+            // }
+            // return <div>Invalid first value: {`${first}`}</div>;
+            return null;
+          }
+        },
+        showIf: (config) =>
+          config.queryFormat.locationType === 'geohash',
+      })
+      .addFieldNamePicker({
+        path: 'fieldMapping.metricField',
+        name: 'Metric Field',
+        defaultValue: defaultOptions.fieldMapping.metricField,
+        settings: {
+          filter: (f) => f.type === FieldType.number,
+          noFieldsMessage: 'No numeric fields found',
+        },
+      })
+      .addSliderInput({
+        path: 'radius',
```

Comment:
```suggestion
        path: 'config.radius',
```
Side effect from: https://github.com/grafana/grafana/pull/36768